bundles/org.openhab.binding.lgwebos/src/main/java/org/openhab/binding/lgwebos/internal/handler/LGWebOSTVSocket.java

```diff
@@ -0,0 +1,796 @@
+        CONNECTING,
+        REGISTERING,
+        REGISTERED,
+        DISCONNECTING
+    }
+
+    private State state = State.DISCONNECTED;
+
+    private final ConfigProvider config;
+    private final WebSocketClient client;
+    private @Nullable Session session;
+    private final URI destUri;
+    private @Nullable WebOSTVSocketListener listener;
+    private final LGWebOSTVKeyboardInput keyboardInput;
+    /**
+     * Requests to which we are awaiting response.
+     */
+    private HashMap<Integer, ServiceCommand<?>> requests = new HashMap<>();
+
+    private final Logger logger = LoggerFactory.getLogger(LGWebOSTVSocket.class);
+    private int nextRequestId = 0;
+
+    private static String FOREGROUND_APP = "ssap://com.webos.applicationManager/getForegroundAppInfo";
+    // private static String APP_STATUS = "ssap://com.webos.service.appstatus/getAppStatus";
+    // private static String APP_STATE = "ssap://system.launcher/getAppState";
+    private static String VOLUME = "ssap://audio/getVolume";
+    private static String MUTE = "ssap://audio/getMute";
+    // private static String VOLUME_STATUS = "ssap://audio/getStatus";
+    private static String CHANNEL_LIST = "ssap://tv/getChannelList";
+    private static String CHANNEL = "ssap://tv/getCurrentChannel";
+    // private static String PROGRAM = "ssap://tv/getChannelProgramInfo";
+    // private static String CURRENT_PROGRAM = "ssap://tv/getChannelCurrentProgramInfo";
+    // private static String THREED_STATUS = "ssap://com.webos.service.tv.display/get3DStatus";
+
+    public LGWebOSTVSocket(WebSocketClient client, ConfigProvider config, String host, int port) {
+        this.config = config;
+        this.client = client;
+        this.keyboardInput = new LGWebOSTVKeyboardInput(this);
+
+        try {
+            this.destUri = new URI("ws://" + host + ":" + port);
+        } catch (URISyntaxException e) {
+            throw new IllegalArgumentException("IP Address or Host provided is invalid: " + host);
+        }
+    }
+
+    public State getState() {
+        return state;
+    }
+
+    private void setState(State state) {
+        State oldState = this.state;
+        if (oldState != state) {
+            this.state = state;
+            Optional.ofNullable(this.listener).ifPresent(l -> l.onStateChanged(oldState, this.state));
+        }
+    }
+
+    public void setListener(@Nullable WebOSTVSocketListener listener) {
+        this.listener = listener;
+    }
+
+    public void clearRequests() {
+        requests.clear();
+    }
+
+    public void connect() {
+        synchronized (this) {
+            if (state != State.DISCONNECTED) {
+                logger.debug("Not trying to connect. Current state is: {}", state);
+                return;
+            }
+            setState(State.CONNECTING);
+        }
+
+        try {
+            this.client.connect(this, this.destUri);
+            logger.debug("Connecting to: {}", this.destUri);
+        } catch (Exception e) {
```

Comment:
can you be more specific in the exception you catch here? (also other occurences in this file)

Reply:
yes, this should be IOException